tests/test_user_security.py

```diff
@@ -225,217 +115,176 @@ def test_has_access(self):
         self.assertTrue(has_access(security_handler.from_session_data(self.update), [User.Status.ACTIVE]))
         self.assertTrue(has_access(security_handler.from_session_data(self.user), [User.Status.ACTIVE]))
 
-    def test_secure_entry(self):
+    def test_session_data(self):
+        time_stamp = datetime.utcnow().timestamp()
+        user = result_to_dict(user_data_access.get(1))
+        user_session = SecurityHandler.get().generate_session_data_token(user)
+        decoded_dict = jwt.decode(user_session, options={"verify_signature": False})
+
+        self.assertIsInstance(decoded_dict, dict)
+        self.assertEqual(decoded_dict['id'], user['id'])
+        self.assertEqual(decoded_dict['email'], user['email'])
+        self.assertGreater(decoded_dict['exp'], time_stamp)
+        self.assertEqual(int(decoded_dict['exp']), int(time_stamp) + 30)
+
+    def test_secure_django(self):
+        # DJANGO INIT
+        if not settings.configured:
+            settings.configure()
+            settings.DEFAULT_CHARSET = 'utf-8'
+        WebHelpersUtils.init(WebHelpersUtils.ServerType.DJANGO)
+
+        self._test_secure_entry()
+        self._test_expiry()
+
+    def test_secure_flask(self):
+        # FLASK INIT
+        WebHelpersUtils.init(WebHelpersUtils.ServerType.FLASK)
+
+        self._test_secure_entry()
+        self._test_expiry()
+
+    def _create_and_send_request(self, payload: dict, policies: list):
+        token = jwt.encode(payload, secret_key)
+        if WebHelpersUtils.get_server_type() == WebHelpersUtils.ServerType.FLASK:
+            header = dump_cookie(cookie_name, token)
+            with self.app.test_request_context(environ_base={'HTTP_COOKIE': header}):
+                self.assertEqual(request.cookies[cookie_name], token)
+                return require_login(request, policies, entry_as_per_policies)
+        else:
+            request_object = HttpRequest
+            request_object.COOKIES = {cookie_name: token}
+            self.assertEqual(request_object.COOKIES[cookie_name], token)

```

Comment:
```
request = None
if Django 
   request = Django request
else: 
   request = flask request 
return require_login(request_object, policies, entry_as_per_policies)
```

Reply:
Hi, Can't take require_login outside the condition for the  reason that we have with statement inside the if condition and we have to call request inside the with context only. Thank you